### PR TITLE
Allow `path` to be used to specify path for synced Realms

### PR DESCRIPTION
**Pull request by kneth**
## ☑️ ToDos

* [x] 📝 Changelog entry
* ~[ ] 📝 `Compatibility` label is updated or copied from previous entry~
* ~[ ] 📝 Update `COMPATIBILITY.md`~
* [x] 🚦 Tests
* [x] 🔀 Executed flexible sync tests locally if modifying flexible sync
* ~[ ] 📦 Updated internal package version in consuming `package.json`s (if updating internal packages)~
* ~[ ] 📱 Check the React Native/other sample apps work if necessary~
* ~[ ] 📝 Public documentation PR created or is not necessary~
* ~[ ] 💥 `Breaking` label has been applied or is not necessary~

*If this PR adds or changes public API's:*
* ~[ ] typescript definitions file is updated~
* ~[ ] jsdoc files updated~

### CODE REVIEW

**Review by takameyer — APPROVED** [review]
Looks good.  Just a couple things.

File: src/js_sync.hpp [hunk 1]
```diff
@@ -1426,7 +1427,17 @@ void SyncClass<T>::populate_sync_config(ContextType ctx, ObjectType realm_constr
         }
 
         config.schema_mode = SchemaMode::AdditiveExplicit;
-        config.path = user->m_user->sync_manager()->path_for_realm(*(config.sync_config));
+        if (config.path.empty()) {
+            config.path = user->m_user->sync_manager()->path_for_realm(*(config.sync_config));
+        }
+        else {
+            // remove .realm if it exists
+            std::size_t found = config.path.rfind(".realm");
```
[the remark this answers is not in the document]
Comment (kneth) [reply src/js_sync.hpp:1435]: `.realm` seems to be added at the end, so if the configuration is `{ path: "foobar.realm", ...}` the file name will become `.../foobar.realm.realm`.

File: integration-tests/tests/src/tests/sync/flexible.ts [hunk 2]
```diff
@@ -121,6 +121,7 @@ async function addSubscriptionAndSync<T>(
 }
 
 describe.skipIf(environment.missingServer, "Flexible sync", function () {
+  this.timeout(20 * 1000);
```
[the remark this answers is not in the document]
Comment (kneth) [reply integration-tests/tests/src/tests/sync/flexible.ts:124]: Hmm, and locally it is set to 2 seconds. But I will remove it.

[the remark this answers is not in the document]
Comment (takameyer) [reply integration-tests/tests/src/tests/sync/flexible.ts:124]: The default timeout is 2 seconds.  We need to come up with a better way to do this, as this keeps coming up 🤷🏼

File: src/js_sync.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (takameyer) [reply src/js_sync.hpp:1435]: What happens if the path is `{ path: "foober", ...}`?  Would the filename be `.../foobar`? (and maybe that's fine)

[the remark this answers is not in the document]
Comment (kneth) [reply src/js_sync.hpp:1435]: For synced Realms, `{ path: "foobar", ... }` will give the file name `"foobar.realm"`.

[the remark this answers is not in the document]
Comment (takameyer) [reply src/js_sync.hpp:1435]: I guess then we should remove `.realm` in the code reading the configuration, before it goes to core (if it exists).  Then the realm file will always get the `.realm` file extension

**Pull request closed without merging**